text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/tag/standard/GradientTag.java

```diff
@@ -51,12 +51,11 @@ final class GradientTag extends AbstractColorChangingTag {
   static final TagResolver RESOLVER = TagResolver.resolver(GRADIENT, GradientTag::create);
 
   private int index = 0;
-  private int colorIndex = 0;
 
-  private float factorStep = 0;
+  private double multiplier = 1;
+
   private final TextColor[] colors;
-  private float phase;
-  private final boolean negativePhase;
+  private @Range(from = -1, to = 1) double phase;
```

Comment:
miniscule nit: here's a place below on line 71 (which github won't let me comment on) where phase is assigned from a double casted to a float which is probably redundant now that phase is a double (altho the information loss is so inconsequential)